server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSConverseOptionsCheckUtility.java

```diff
@@ -82,31 +82,34 @@ public CICSConverseOptionsCheckUtility(
      * @param <E> A subclass of ParserRuleContext
      */
     public <E extends ParserRuleContext> void checkOptions(E ctx) {
-        int currIndex = ctx.getRuleIndex();
-        switch (currIndex) {
-            case CICSParser.RULE_cics_converse_group_one:
-                checkGroupOne((CICSParser.Cics_converse_group_oneContext) ctx);
-                break;
-            case CICSParser.RULE_cics_converse_group_two:
-                checkGroupTwo((CICSParser.Cics_converse_group_twoContext) ctx);
-                break;
-            default:
-                break;
+        if (ctx.getRuleIndex() == CICSParser.RULE_cics_converse_group) {
+            checkRule((CICSParser.Cics_converse_groupContext) ctx);
         }
     }
 
-    private void checkGroupOne(CICSParser.Cics_converse_group_oneContext ctx) {
+    private void checkRule(CICSParser.Cics_converse_groupContext ctx) {
         if (!ctx.ASIS().isEmpty()) {
             checkHasIllegalOptions(ctx.LEAVEKB(), "LEAVEKB");
         }
-        checkHasMandatoryOptions(ctx.FROM(), ctx, "FROM");
-        checkHasMandatoryOptions(ctx.cics_into(), ctx, "INTO or SET");
 
-        checkDuplicates(ctx);
-    }
+        if (!ctx.cics_converse_fromlength().isEmpty()) {
+            checkHasMandatoryOptions(ctx.FROM(), ctx, "FROM");
+        }
+
+        if (!ctx.CTLCHAR().isEmpty()) {
+            checkHasIllegalOptions(ctx.CONVID(), "CONVID");
+            checkHasIllegalOptions(ctx.STRFIELD(), "STRFIELD");
+        }
+
+        if (!ctx.cics_converse_erase().isEmpty()) {
+            checkHasIllegalOptions(ctx.STRFIELD(), "STRFIELD");
+        }
+        if (!ctx.STRFIELD().isEmpty()) {
+            checkHasIllegalOptions(ctx.cics_converse_erase(), "CTLCHAR");
+        }
 
-    private void checkGroupTwo(CICSParser.Cics_converse_group_twoContext ctx) {
         checkHasMandatoryOptions(ctx.cics_into(), ctx, "INTO or SET");
+        checkHasMandatoryOptions(ctx.cics_converse_tolength(), ctx, "TOLENGTH OR TOFLENGTH");
```

Comment:
isn't a mandatory options when they used with FROMLENGTH even if they are shown mandatory on docs. 
![image](https://github.com/user-attachments/assets/aeb5cf8d-46e4-45c3-aedf-8b8830fb51f8)

however they are mandatory when used with MAXFLENGTH or MAXLENGTH  them like this case;

![image](https://github.com/user-attachments/assets/4bccc7e0-cc26-4a5d-9f14-73921cd55b97)